Support for a heartbeat status check.

lokkilok opened the pull request:
iThe status check should be defined in the db like this:
"status" : {
        "type" : "Heartbeat",
        "count" : 2,
        "minutes" : 5
    }

Commit:
Support for a heartbeat status check. The status check should be defined
in the db like this: 	"status" : {
		"type" : "Heartbeat",
		"count" : 2,
		"minutes" : 5
	}

PoncinMatthieu commented on app/model/service.py:
```diff
@@ -131,14 +131,28 @@ def TryCheckStatus(self):
+						'time': {'$gt': datetime.datetime.now() - datetime.timedelta(minutes=period)}}
+				if tag:
+					query['tags'] = tag
+				count = Event.Count(self.id, query)
```
Might want to be very careful with this, this could be super slow on the prod db.
It's all fine for a start, but I think it would make more sense to check those heartbeats when actually receiving the events (maybe in the event class on insert?)

PoncinMatthieu replied:
Yeah this method is called every minutes so i'm not sure to understand how it works (but i'm also a bit tired atm, i can have a better look tomorrow morning :D )

lokkilok replied:
Yes without the right index(es) on the events could be very slow if there are many, many events.
An alternative would be to have a little cache (a queue) for "heartbeat" events that has e.g. the last 10 or so of them, but then that cache should be shared all Monni processes. A cache that is shared between processes is a db by another name, so.... Perhaps a good solution is to have a dedicated, and capped, collection for heartbeats, so that the query in  the status check will be fast, without requiring that the index needs to be large.

Commit:
Reworked heartbeat support such that it's now possible to configure a
dedicated collection for certain event types. So for heartbeats it's
recommended to configure a dedicated collection which is either capped,
or has a automatic expiration index.

lokkilok commented:
So the latest commit allows for dedicated collections.... That should be ok performance wise, as long as such a heartbeat collection is properly set up.

faltad committed:
Adding the previous url while redirecting after login

Commit:
Merge pull request #1 from lokkilok/feature/betterloginredirect

Adding the previous url while redirecting after login

PoncinMatthieu commented:
ok at least the dedicated heartbeat collection won't slow down anything for sure
